homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -822,10 +826,24 @@ async def async_update(self):
                 }
             )
 
+            self._retry = 0
+
         except DeviceException as ex:
-            if self._available:
-                self._available = False
-                _LOGGER.error("Got exception while fetching the state: %s", ex)
+            self._retry += 1
+            if self._retry < self._retries:
+                _LOGGER.info(
+                    "Got exception while fetching the state: %s, retry: %d",
+                    ex,
+                    self._retry,
+                )
```

Comment:
Where is the retry mechanism?

Reply:
Then I'd rename the attributes to `failures` or similar to signify that we allow a number of failures until marking the entity as unavailable. We don't retry explicitly.

Reply:
Yes, of course.